File: solutions/python/darts/1/darts.py
def score(x: int, y: int) -> int:
    distance = (x**2 + y**2)**.5
    if distance <= 1:
        return 10
    if 1 < distance <= 5:
        return 5
    if 5 < distance <= 10:
        return 1
    return 0

File: solutions/ruby/darts/1/darts.rb
class Darts
  attr_reader :score

  def initialize(x, y)
    @score = compute_score(x, y)
  end

  def compute_score(x, y)
    case (x**2+y**2)**0.5
    when 0..1; 10
    when 1..5; 5
    when 5..10; 1
    else 0
    end
  end
end
